Gemfile.lock

```diff
@@ -330,9 +349,10 @@ GEM
     zeitwerk (2.4.2)
 
 PLATFORMS
-  ruby
+  x86_64-linux
 
```

Comment:
Can you help me understand if Gemfile.lock should be included in our git commits or is it okay to skip this? Would a deployment platform like Heroku not build this anyways assuming they would run `bundle install` when building the app?

Reply:
Other issues we have started noticing is that when there are developers some of whom have the old Macbooks with Intel chips and the new Macbooks with M1 chips, the x86_64 gems start causing issues on the M1 arm architecture and vice versa. One workaround we could figure out was when developers run `git pull` on the `main` branch, if they simply delete the `Gemfile.lock` it allows rails to install the correct dependencies. This is why we have been questioning if `Gemfile.lock` should be added to the `.gitignore` file. Wondering if you encountered such issues when collaborating on Rails projects?